tests/bugs/ptr-extension-this-type.slang

```diff
@@ -0,0 +1,25 @@
+//TEST:COMPILE: -target hlsl
+
+// Test for issue #7656: Compiler segfault with Ptr<T> extension referring to This
+// This test ensures that the compiler doesn't segfault when 'This' is used in Ptr<T> extensions
+
+public interface IThing
+{
+    void f(This other);
+}
+
+public extension Ptr<int>: IThing
+{
+    public void f(This other)
+    {
+        // This function should compile without segfault
```

Comment:
maybe change this function to `return sizeof(This) - sizeof(Ptr<int>) + 1;` and run the test using the interpreter to verify correctness? should expect it to be equal to `1`.